tests/e2e/lib/virt_helpers.go

```diff
@@ -513,6 +581,155 @@ func (v *VirtOperator) ensureHcoRemoved(timeout time.Duration) error {
 	return err
 }
 
+func (v *VirtOperator) getVmStatus(namespace, name string) (string, error) {
+	vm, err := v.Dynamic.Resource(virtualMachineGvr).Namespace(namespace).Get(context.Background(), name, v1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	status, ok, err := unstructured.NestedString(vm.UnstructuredContent(), "status", "printableStatus")
+	if err != nil {
+		return "", err
+	}
+	if !ok {
+		return "", fmt.Errorf("status field not populated yet on VM %s/%s", namespace, name)
+	}
+	log.Printf("VM %s/%s status is: %s", namespace, name, status)
+
+	return status, nil
+}
+
+func (v *VirtOperator) checkVmExists(namespace, name string) bool {
+	_, err := v.getVmStatus(namespace, name)
+	if err == nil {
+		return true
+	}
+	return false
+}
+
+func (v *VirtOperator) checkVmStatus(namespace, name, expectedStatus string) bool {
+	status, _ := v.getVmStatus(namespace, name)
+	return status == expectedStatus
+}
+
+func (v *VirtOperator) createVm(namespace, name, source string) error {
+	unstructuredVm := unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "kubevirt.io/v1",
+			"kind":       "VirtualMachine",
+			"metadata": map[string]interface{}{
+				"name":      name,
+				"namespace": namespace,
+			},
+			"spec": map[string]interface{}{
+				"running": true,
+				"template": map[string]interface{}{
+					"spec": map[string]interface{}{
+						"domain": map[string]interface{}{
+							"devices": map[string]interface{}{
+								"disks": []map[string]interface{}{
+									{
+										"disk": map[string]interface{}{
+											"bus": "virtio",
+										},
+										"name": "rootdisk",
+									},
+								},
+							},
+							"resources": map[string]interface{}{
+								"requests": map[string]interface{}{
+									"cpu":    "1",
+									"memory": "256Mi",
+								},
+							},
+						},
+						"volumes": []map[string]interface{}{
+							{
+								"dataVolume": map[string]interface{}{
+									"name": name,
+								},
+								"name": "rootdisk",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if _, err := v.Dynamic.Resource(virtualMachineGvr).Namespace(namespace).Create(context.TODO(), &unstructuredVm, v1.CreateOptions{}); err != nil {
+		return fmt.Errorf("error creating VM %s/%s: %w", namespace, name, err)
+	}
+
+	return nil
+}
+
+func (v *VirtOperator) removeVm(namespace, name string) error {
+	if err := v.Dynamic.Resource(virtualMachineGvr).Namespace(namespace).Delete(context.TODO(), name, v1.DeleteOptions{}); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("error deleting VM %s/%s: %w", namespace, name, err)
+		}
+		log.Printf("VM %s/%s not found, delete not necessary.", namespace, name)
+	}
+
+	return nil
+}
+
+func (v *VirtOperator) ensureVm(namespace, name, source string, timeout time.Duration) error {
```

Comment:
here the logic is right? It will check if VM exists (without checking status), but if it fails, then it will check status

Reply:
Maybe not, it doesn't look like I thought this through. The existence check is really there to speed up development of the CI test itself, since I don't have it tear down and recreate everything from scratch on every run. The status check polling is for the newly created VM. I can probably take out the existence check, since Kubernetes will complain if the VM is already there when the test tries to create it.